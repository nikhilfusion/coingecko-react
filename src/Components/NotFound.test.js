import React from 'react';
import NotFound from './NotFound';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

test("NotFound renders with correct text", () => {
  const { getByTestId } = render(<NotFound />);
  const headerEl = getByTestId("title");
  expect(headerEl.textContent).toBe("404: The page you are looking for isnâ€™t here");
})

test("Button text", () => {
  render(<NotFound />);
  expect(screen.getByRole('button')).toHaveTextContent('Back to home');
})

test('it renders correctly', () => {
    const tree = renderer.create(<NotFound />).toJSON();
    expect(tree).toMatchSnapshot()
})
