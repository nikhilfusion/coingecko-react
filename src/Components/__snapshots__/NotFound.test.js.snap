// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <h1
    className="MuiTypography-root MuiTypography-h1 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
    data-testid="title"
  >
    404: The page you are looking for isnâ€™t here
  </h1>
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
  >
    You either tried some shady route or you came here by mistake. Whichever it is, try using the navigation.
  </h6>
  <div
    className="MuiBox-root MuiBox-root-11"
  >
    <img
      alt="Under development"
      className="makeStyles-image-10"
      src="not_found.svg"
    />
  </div>
  <div
    className="MuiBox-root MuiBox-root-12"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Back to home
      </span>
    </button>
  </div>
</div>
`;
